lightyear/src/server/networking.rs

```diff
@@ -295,15 +312,59 @@ pub(crate) fn send(
                     metrics::counter!("transport::send::packets").increment(packets);
                     metrics::counter!("transport::send::kb").increment(bytes);
                 }
-                netserver.send(packet_byte.as_slice(), *client_id)?;
+                if let Err(e) = netserver.send(packet_byte.as_slice(), *client_id) {
+                    send_errors.push((*client_id, netserver_idx, e));
+                }
             }
             Ok(())
         })
         .unwrap_or_else(|e: ServerError| {
             error!("Error sending packets: {}", e);
         });
+
+    for (client_id, server_idx, error) in send_errors {
+        if let Some(server) = netservers.servers.get_mut(server_idx) {
+            react_to_client_error(&mut connection_manager, server, error);
+        } else {
+            error!("Could not find server for index {} when handling error", server_idx);
+        }
+    }
 }
 
+fn react_to_client_error(
+    connection_manager: &mut ResMut<ConnectionManager>,
+    server: &mut ServerConnection,
+    error: ConnectionError
+ ) {
+    let fatal_error_client_info = match &error {
```

Comment:
Thumbing through the error tree to dig out fatal errors sucks.

Reply:
Yes, ideally we could create a trait `Recoverable` and make the various errors 
(IoError, NetcodeError, TransportError) implement it

Reply:
i'm gonna punt on that until we've identified more than one fatal error, if that's okay